lib/mozilla-ast.js

```diff
@@ -446,8 +446,22 @@
                 args.value = val;
                 return new AST_String(args);
               case "number":
-                args.value = val;
-                return new AST_Number(args);
+                if (isNaN(val)) return new AST_NaN(args);
+                var negate, node;
+                if (isFinite(val)) {
+                    negate = 1 / val < 0;
+                    args.value = negate ? -val : val;
+                    node = new AST_Number(args);
+                } else {
+                    negate = val < 0;
+                    node = new AST_Infinity(args);
+                }
+                return negate ? new AST_UnaryPrefix({
+                    start: args.start,
+                    end: args.end,
+                    operator: "-",
+                    expression: node,
+                }) : node;
```

Comment:
> But I can't isolate the `test/release/mathjs.sh -p acorn --toplevel -mc` failure in `test/unit-tests/type/numeric.test.js`.

I believe this may fix it, but I haven't verified it yet since I'm typing from my Windows box atm...

Reply:
That did the trick. 👍